queries.sql

```diff
@@ -1,10 +1,129 @@
-/*Queries that provide answers to the questions from all projects.*/
-
-SELECT * FROM animals WHERE name LIKE '%mon';
-SELECT * FROM animals WHERE date_of_birth BETWEEN '1-1-2016' AND '31-12-2019';
-SELECT * FROM animals WHERE neutered = '1' AND escape_attempts < 3;
-SELECT date_of_birth FROM animals WHERE name IN ('Agumon', 'Pikachu');
-SELECT name, escape_attempts FROM animals WHERE weight_kg > 10.5;
-SELECT * FROM animals WHERE neutered = '1';
-SELECT * FROM animals WHERE name NOT LIKE 'Gabumon';
-SELECT * FROM animals WHERE weight_kg >= 10.4 AND weight_kg <= 17.3;
+BEGIN;
+
+-- Start a Transaction:
+BEGIN -- Update "species" to "unspecified" for Animals with Names Ending in "mon":
+UPDATE
+  animals
+SET
+  species = 'unspecified'
+WHERE
+  name LIKE '%mon';
+
+UPDATE
+  6 -- Display changes to comfirm
+SELECT
+  *
+FROM
+  animals;
+
+-- Update "species" to "pokemon" for Animals without a Species:
+UPDATE
+  animals BEGIN;
+
+UPDATE
+  animals
+SET
+  species = 'digimon'
+WHERE
+  name LIKE '%mon';
+
+COMMIT;
+
+SET
+  species = 'pokemon'
+WHERE
+  species IS NULL;
+
+UPDATE
+  5 -- TAKE A BREATH 
+  BEGIN;
+
+-- Display changes to comfirm
+SELECT
+  *
+FROM
+  animals;
+
+BEGIN -- Delete Animals Born After Jan 1st, 2022:
+DELETE FROM
+  animals
+WHERE
+  date_of_birth > '2022-01-01';
+
+DELETE 1 SAVEPOINT before_weight_update;
+

```

Comment:
- [ ] Before this query you are missing one query that was about `deleting all records in the animal's table, then rolling back the transaction

```sql
BEGIN;

   DELETE FROM animals WHERE ;
   SELECT COUNT(*) FROM ANIMALS;

   ROLLBACK;
   SELECT COUNT(*) FROM ANIMALS;
```

Reply:
First changes requested is already implemented right beneath it. Sorry is the real difference between 'SELECT * FROM animals' and 'SELECT species FROM animals'??

Reply:
What is the real difference between SELECT * FROM animals and SELECT species FROM animals please??. Difference command similar outcome.